addOns/automation/src/main/java/org/zaproxy/addon/automation/tests/UrlPresenceTest.java

```diff
@@ -0,0 +1,482 @@
+            String onFail,
+            String operator,
+            String url,
+            String requestHeaderRegex,
+            String requestBodyRegex,
+            String responseHeaderRegex,
+            String responseBodyRegex) {
+        LinkedHashMap<String, Object> map = new LinkedHashMap<>();
+        map.put(PARAM_NAME, name);
+        map.put(PARAM_ON_FAIL, onFail);
+        map.put(PARAM_OPERATOR, operator);
+        map.put(PARAM_URL, url);
+        map.put(PARAM_REQUEST_HEADER_REGEX, requestHeaderRegex);
+        map.put(PARAM_REQUEST_BODY_REGEX, requestBodyRegex);
+        map.put(PARAM_RESPONSE_HEADER_REGEX, responseHeaderRegex);
+        map.put(PARAM_RESPONSE_BODY_REGEX, responseBodyRegex);
+        return map;
+    }
+
+    public UrlPresenceTest(
+            String name,
+            String onFail,
+            String operator,
+            String url,
+            String requestHeaderRegex,
+            String requestBodyRegex,
+            String responseHeaderRegex,
+            String responseBodyRegex,
+            AutomationJob job,
+            AutomationProgress progress) {
+        this(
+                paramsToData(
+                        name,
+                        onFail,
+                        operator,
+                        url,
+                        requestHeaderRegex,
+                        requestBodyRegex,
+                        responseHeaderRegex,
+                        responseBodyRegex),
+                job,
+                progress);
+    }
+
+    public UrlPresenceTest(AutomationJob job, AutomationProgress progress)
+            throws IllegalArgumentException {
+        super("", AbstractAutomationTest.OnFail.INFO.name(), job);
+        data = new Data(this);
+        data.setOnFail(AbstractAutomationTest.OnFail.INFO);
+    }
+
+    @Override
+    public String getTestType() {
+        return TEST_TYPE;
+    }
+
+    public static boolean isNullOrEmpty(String str) {
+        return str == null || str.isEmpty();
+    }
+
+    public static boolean isNull(Object obj) {
+        return obj == null;
+    }
+
+    private Pattern compilePattern(String regex, AutomationProgress progress) {
+        try {
+            return Pattern.compile(regex);
+        } catch (PatternSyntaxException e) {
+            progress.warn(
+                    Constant.messages.getString(
+                            "automation.tests.url.badRegex",
+                            getJobType(),
+                            regex,
+                            getName(),
+                            e.getMessage()));
+        }
+        return null;
+    }
+
+    public boolean checkPattern(
+            String regex, String msg, AutomationProgress progress, String operator) {
+        if (isNull(regex)) {
+            return false;
+        }
+        if (isNullOrEmpty(regex) && (operator.equalsIgnoreCase("or") || isNullOrEmpty(operator))) {
+            return false;
+        }
```

Comment:
The above tests dont really make sense to me - I'm not sure why they are there.
It also looks to me like there are some side effects going on here - eg are we delibrately creating a Pattern for an empty (but not null) regex when the operator is "and" ?

Reply:
Bump.